jupyterhub_moss/spawner.py

```diff
@@ -79,11 +82,80 @@ def _validate_partitions(self, proposal):
         return partitions
 
     slurm_info_cmd = traitlets.Unicode(
-        # Get number of nodes and cores for all partitions
-        r"sinfo -a -N --noheader -o '%R %t %m'",
+        # Get number of nodes, cores, gpus, total memory for all partitions
+        r"sinfo -a --noheader -o '%R %D %C %G %m'",
         help="Command to query cluster information from Slurm. Formatted using req_xyz traits as {xyz}.",
     ).tag(config=True)
 
+    slurm_info_resources = traitlets.Callable(
+        help="""Provides information about resources in Slurm cluster.
+
+        It will be called with the Spawner as a single argument and return a tuple:
```

Comment:
Shouldn't this be:
```suggestion
        It will be called with the output of ``slurm_info_cmd`` as a single argument and return a tuple:
```
or
 ```suggestion
        It will be called with the Spawner and the output of ``slurm_info_cmd`` as arguments and return a tuple:
```

Reply:
I'll expand this docstring, it can be improved

Reply:
done in https://github.com/silx-kit/jupyterhub_moss/pull/66/commits/c317404719e4eaf596453f41577a2fc75610cd98